sound/hybrid_font.h

```diff
@@ -234,11 +234,38 @@ class HybridFont : public SaberBase, public Looper {
     EnableAmplifier();
     if (!f->files_found()) return RefPtr<BufferedWavPlayer>(nullptr);
     RefPtr<BufferedWavPlayer> player = GetFreeWavPlayer();
-    if (player) {
-      player->set_volume_now(font_config.volEff / 16.0f);
-      player->PlayOnce(f);
-      current_effect_length_ = player->length();
+    if (!player) {
+      STDOUT.println("Out of WAV players! Getting more...");
+      float oldest_ = 0;
+      uint32_t kill_ = 0;
+      // check wavplayers, rule out hum and smoothswings then kill oldest one 
+      for (size_t i = 0; i < NELEM(wav_players); i++) {
+        if (wav_players[i].isPlaying() && wav_players[i].current_file_id().GetEffect() == f) {
+          float pos = wav_players[i].pos();
+          if (pos > oldest_) {
+            oldest_ = pos;
+            kill_ = i;
+          }    
+        } else {
+          // unsuccessful acquisition. Let's try again.
+          if (i == 6) return player;
```

Comment:
this is the wrong end condition
also, we should never hardcode a number like this, as it would stop working if we ever change the number of wav players.

Reply:
Good point on the future.
If not returning, what's more appropriate? It did not like break;

Reply:
this code just needs to go away
after the loop we should check if kill_ (or whatever rename that variable to) is -1.

Reply:
changed.

Reply:
Ok I've found the fail point in this for-loop. I think we just need to check if the sound  is *an* effect, instead of if it is the *currect* effect.  
`wav_players[i].current_file_id().GetEffect() == e` will fail if we're loaded up with blst wavs, then a clsh comes through.
This is where it freezes because this scenario is not covered.  
How could we broaden the filter so any file NOT hum smoothswing or track is ok?

Reply:
I got it! We check with GetEffect() but don't compare to *current* effect.
Working well.
Removed time left for fade out check, unless that's needed. I think it goes away.
Removed fail case for the for-loop as it will always get a player playing an effect now.
I still don't get what we would need to return in order to fail gracefully,  ie: not get a player to free up, but still satisfy the function somehow. But not failing seems like a good idea.

Reply:
> 

What would we do if it were still -1 because for SOME reason we got no players to kill?
Looking for what we would end up returning to the function.

Reply:
If we don't check that it's the same effect, then things gets complicated, because we should never kill the hum, humm, in, out, track, swingl, swingh, and probably a lot of other things. Basically it gets hard to know what we can/should kill.

Also, the assertion that we should never see a case where we can't kill an old wav player is just not correct.
However, this particular if statement should still go, it was never working the way you meant to anyways.

Reply:
Yes, this particular if statement was wrong.
Currently though, it's working. I tested it hard, with a printout for every step and it was all happy throughout, working as expected.
I did only test spamming blaster blocks mixed with clashes. Both were recognized as Effects, while hum smoothswings and track playing were ignored.
Not sure how out.wav or in.wav would get in the way. Maybe though.
I've hit a wall on what I can discover throughout the OS as far as what else to use to filter results to be NOT be a sound on the bad list you wrote above.